conans/model/manifest.py

```diff
@@ -108,17 +98,23 @@ def create(cls, folder, exports_sources_folder=None):
         from disk, and capturing current time
         """
         files, _ = gather_files(folder)
+        # The folders symlinks are discarded for the manifest
         for f in (PACKAGE_TGZ_NAME, EXPORT_TGZ_NAME, CONAN_MANIFEST, EXPORT_SOURCES_TGZ_NAME):
             files.pop(f, None)
 
         file_dict = {}
         for name, filepath in files.items():
-            file_dict[name] = md5sum(filepath)
+            # For a symlink: md5 of the pointing path, no matter if broken, relative or absolute.
+            value = md5(os.readlink(filepath)) if os.path.islink(filepath) else md5sum(filepath)
```

Comment:
why? this would be pretty useless to the manifest, if the file is not packaged, why computing its md5? Why not just blank or the ``md5`` of the ``source`` path (not the target one).

Reply:
Well, I don't think is useless. The file is the symlink itself, and the content is the path where it is pointing. I think it is relevant if you are packaging a different symlink, pointing to a different location.

Reply:
But that is the thing. If you have a ``pkg/lib/somelib.so -> /usr/lib/somelib.so``, then the system ``somelib.so`` might change from computer to computer, so the Manifest ``conan_manifest.txt`` will be broken, with an incorrect md5, and an irreproducible/uncheckable/wrong package revision. The idea is that we can validate the contents of a Conan package validating its manifest.

Reply:
The manifest won't be broken, it will be exactly the same, encoding the string `/usr/lib/somelib.so`, if the symlinks point to the system, and the system changes, of course, the final package will be different, in terms of linking with another different file, but it will be different anyway even if you hash an empty string.
So, if the manifest "is broken" in this case, then is "broken" in any case with external symlinks.

Reply:
Oh, I see! I was confused by ``md5`` vs ``md5sum``. 
Now makes sense, thanks.